### DISCUSSION TITLE
Vmap with index

### DISCUSSION THREAD

**Discussion by jakubMitura14** [jax-ml/jax#13469] [Q&A] [answered]
Hello I have 3 dimensional dense array I want to perform some stencil operation that will require both information about Cartesian coordinates of the point and about the neighberhood 

If I understand correctly I should be able to write the operation as if working on single voxel and vmap it over all array my problem is that I do not see how to get access to neighberhood  and Cartesian coordinates and to vmap over 3 dim array

Is it possible to use vmap in a way I described?

**Comment by jakevdp** [comment, marked as answer]
`vmap` is essentially an elementwise operation. So if you can write your operation in a form that looks roughly like this:
```python
output = []
for elem in arr:
  output.append(do_something(elem))
result = jnp.stack(output)
```
then you can use `result = vmap(do_something)(arr)` to achieve this with much better performance.

When you mention "neighborhood", it sounds like you're talking about an operation that does not operate element-wise, and if that is the case you won't be able to directly express it using `vmap`. If you can edit your question to show a more concrete example of the type of operation you're trying to do, we may be able to give more specific help.

**Reply by jakubMitura14** [reply]
Thanks for answer!! Yes as the neughberhood I mean voxel that is next previous below and up , so I suppose it is impossible via naive implementation, and is it possible to keep access to 3d index I an array ? I mean x,y, z

**Reply by jakevdp** [reply]
It sounds like you're talking about a convolution. Perhaps you could express your computation in terms of `jax.scipy.signal.convolve` or something similar?

**Reply by jakubMitura14** [reply]
Thanks this is for sure path worth investigating !
